cli/src/main.rs

```diff
@@ -853,17 +853,26 @@ impl Parse {
 
         let timeout = self.timeout.unwrap_or_default();
 
-        let (paths, language) = if let Some(target_test) = self.test_number {
+        let (paths, language, parse_stdin) = if let Some(target_test) = self.test_number {
             let (test_path, language_names) = test::get_tmp_test_file(target_test, color)?;
             let languages = loader.languages_at_path(current_dir)?;
             let language = languages
                 .iter()
                 .find(|(_, n)| language_names.contains(&Box::from(n.as_str())))
                 .map(|(l, _)| l.clone());
-            let paths = collect_paths(None, Some(vec![test_path.to_str().unwrap().to_owned()]))?;
-            (paths, language)
+            let paths =
+                collect_paths(None, Some(vec![test_path.to_str().unwrap().to_owned()]))?.unwrap();
+            (paths, language, false)
         } else {
-            (collect_paths(self.paths_file.as_deref(), self.paths)?, None)
+            (
+                if let Some(collected) = collect_paths(self.paths_file.as_deref(), self.paths)? {
+                    collected
+                } else {
+                    vec![get_tmp_parse_file()?.to_str().unwrap().to_owned()]
+                },
+                None,
+                true,
+            )
```

Comment:
can you hoist out the paths into its own variable above the tuple construction? This feels a bit unruly and messy...also, there's a repeated habit of calling `get_tmp_parse_file()?.to_str().unwrap().to_owned()`, and `get_tmp_parse_file()` isn't using its path besides to convert it as a string, so we should move that logic into `get_tmp_parse_file` and have it return a String, so in the end we have something like this:

```suggestion
            let paths =
                if let Some(collected) = collect_paths(self.paths_file.as_deref(), self.paths)? {
                    collected
                } else {
                    vec![get_tmp_parse_file()?]
                };
            (paths, None, true)
```